backend/controller/artefacts/dal_registry.go

```diff
@@ -0,0 +1,80 @@
+package artefacts
+
+import (
+	"context"
+	"github.com/TBD54566975/ftl/backend/libdal"
+	"github.com/TBD54566975/ftl/internal/sha256"
+	"github.com/TBD54566975/ftl/internal/slices"
+	sets "github.com/deckarep/golang-set/v2"
+	"io"
+)
+
+type ArtefactRow struct {
+	ID     int64
+	Digest []byte
+}
+
+type DAL interface {
+	GetArtefactDigests(ctx context.Context, digests [][]byte) ([]ArtefactRow, error)
+	CreateArtefact(ctx context.Context, digest []byte, content []byte) (int64, error)
+	GetArtefactContentRange(ctx context.Context, start int32, count int32, iD int64) ([]byte, error)
+}
+
+type DALRegistry struct {
+	db DAL
+}
+
+func NewDALRegistry(db DAL) *DALRegistry {
+	return &DALRegistry{db: db}
+}
+
+func MakeKey(id int64, digest []byte) ArtefactKey {
```

Comment:
Same as other comment, we shouldn't need to do this.

Reply:
yep, removed